finalitygadget/finalitygadget.go

```diff
@@ -499,43 +574,116 @@ func (fg *FinalityGadget) queryBlockByHeight(blockNumber int64) (*types.Block, e
 	}, nil
 }
 
-func (fg *FinalityGadget) handleBlock(ctx context.Context, latestFinalizedHeight uint64) error {
-	for height := fg.lastProcessedHeight + 1; height <= latestFinalizedHeight; height++ {
+type blockResult struct {
```

Comment:
i had a hard time following this struct and related logic. my understanding:

- `blockResult` is mostly only used as the return type for `processHeight`
- `processHeight` is only called by `processBlocksTillHeight`
- the `block` field is never used if it's not finalized

I think we can completely get rid of this struct to make the logic cleaner:
- refactor `processHeight` to `processBlock(block *types.Block) (error, bool)`
- if we don't want to add another go routine for batch calling `queryBlockByHeight()`, we can refactor the function to `processHeightAndReturnIfFinalized(height uint64) (error, *types.Block)`
- if it's not finalized, it will return `nil, nil`

Reply:
updated for this in latest commit